File: sdk/javascript/src/instrumentation/openai.ts
/**
 * OpenAIInstrumentor — automatic usage capture for the official `openai`
 * npm package's Chat Completions and Responses APIs (streaming and
 * non-streaming).
 *
 *     import OpenAI from "openai";
 *     import { OpenAIInstrumentor } from "@costorah/sdk";
 *
 *     new OpenAIInstrumentor().instrument();
 *
 *     const client = new OpenAI();
 *     await client.responses.create({ model: "gpt-4.1", input: "Hello" });
 */

import { OpenAICompatibleInstrumentor } from "./openaiCompatible.js";

export class OpenAIInstrumentor extends OpenAICompatibleInstrumentor {
  readonly name = "openai";
  readonly fixedProvider = "openai";
}

File: sdk/python/costorah/instrumentation/openai.py
"""
OpenAIInstrumentor — automatic usage capture for the official `openai`
Python package's Chat Completions and Responses APIs (sync, async, and
streaming).

    from openai import OpenAI
    from costorah.instrumentation import OpenAIInstrumentor

    OpenAIInstrumentor().instrument()

    client = OpenAI()
    client.responses.create(model="gpt-4.1", input="Hello")

See `_openai_compatible.py` for the shared patch implementation this (and
every other OpenAI-SDK-compatible provider instrumentor) is built on.
"""

from __future__ import annotations

from costorah.instrumentation._openai_compatible import _OpenAICompatibleInstrumentor


class OpenAIInstrumentor(_OpenAICompatibleInstrumentor):
    name = "openai"
    fixed_provider = "openai"
